starter_code/elevator.js

```diff
@@ -26,44 +26,69 @@ class Elevator {
     stop() { 
       this.elevatorRunning = false;
     }
+
     update() {
+      if (this.passengers.length != 0) {
+          if (this.passengers[0].destinationFloor > this.floor) {
+            this.floorUp();
+          } else if (this.passengers[0].destinationFloor < this.floor) {
+            this.floorDown();
+          } 
+      } else { 
+        if (this.requests[0] > this.floor) {
+          this.floorUp();
+        } else if (this.requests[0] < this.floor) {
+          this.floorDown();
+        }
+      }
       this.log();
+      if (this.requests.indexOf(this.floor) != -1) {
+        this.requests.splice(this.requests.indexOf(this.floor), 1);
+      }
     }
-    _passengersEnter() { }
-    _passengersLeave() { }
+
+    _passengersEnter() {
+      var newPassengers = this.waitingList.filter(item => (item.originFloor == this.floor));
+      var liftPassengers = this.passengers.concat(newPassengers);
+      this.passengers = liftPassengers;
+      newPassengers.map(item => console.log(`${item.name} has joined the elevator.`));
+      this.waitingList = this.waitingList.filter(item => (item["originFloor"] != this.floor));
+     }
+
+    _passengersLeave() { 
+      this.passengers.filter(item => item["destinationFloor"] == this.floor).map(item => console.log (`${item.name} has left the elevator`));
+      this.passengers = this.passengers.filter(item => item["destinationFloor"] != this.floor);
+      
+    }
+
     floorUp() {
       this.direction = "Up";
       if (this.floor < this.MAXFLOOR) {
         this.floor++;
       }
+      this._passengersEnter();
```

Comment:
Maybe you can call this methods in update instead of repeating them in floorUp() and floorDown().